core/src/main/java/com/opensymphony/xwork2/security/DefaultExcludedPatternsChecker.java

```diff
@@ -81,7 +90,14 @@ public void setExcludedPatterns(String[] patterns) {
     }
 
     public void setExcludedPatterns(Set<String> patterns) {
-        LOG.trace("Sets excluded patterns [{}]", patterns);
+        if (excludedPatterns != null && excludedPatterns.size() > 0) {
+            LOG.warn("Replacing excluded patterns [{}] with [{}], be aware that this affects all instances and safety of your application!",
+                        excludedPatterns, patterns);
+        }
+        else {
+            LOG.debug("Replacing excluded patterns [{}] with [{}], be aware that this affects all instances and safety of your application!",
```

Comment:
It should be `LOG.debug("Sets excluded patterns [{}]", patterns);`.

Reply:
Hello @yasserzamani .
The two `debug` log statements have been updated in the most recent commit based on your comments/request.  I left in the extra warning text information but the wording and parameters of the start of the log output match what you requested.  :)

Reply:
Wow! thank you. Now I understand what those tests test :)

Regarding my suggestion on debug log statements, I actually meant for example when acceptedPaterns is null and is going to set to a not-null value, then user doesn't need to `be aware that this affects all instances and safety of your application!` because this increases safety. I afraid such message confuses user. this is same for my other suggestions about debug messages in this PR and that another PR. Please excuse me for my obsession :)

Reply:
Hello @yasserzamani .
The latest commit should update the `debug` statements to match what you requested.  The _accepted patterns_ debug statement still provides a milder notice since adding accepted patterns does potentially impact/reduce safety (in contrast to excluded patterns).
Hope it looks OK now.  :)

Reply:
Yes it's great. thank you :)